x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/PutEngineAction.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class PutEngineAction extends ActionType<PutEngineAction.Response> {
+
+    public static final PutEngineAction INSTANCE = new PutEngineAction();
+    public static final String NAME = "cluster:admin/engine/put";
```

Comment:
Since we require the manage index privilege for the creation of the alias we should use `"indices:admin/engine/put"`. That way we don't require another privilege for engine management. Managements of engines should be done under the manage index privilege. 
The advantage is  that we get the resolution of index pattern (`index-*`) to respect the underlying security.
Indices resolution happens at the transport level on the `IndicesRequest`. So instead of `ActionRequest`, the `Request` below should extends `IndicesRequest` and get/set the list indices from the `IndicesRequest::indices` function.

Reply:
Also note that the tests are failing due to:
```
 Actions are neither operator-only nor non-operator: [[cluster:admin/engine/put]]. An action should be declared to be either operator-only in [org.elasticsearch.xpack.security.operator.OperatorOnlyRegistry] or non-operator in [org.elasticsearch.xpack.security.operator.Constants
```
The new action needs to be added in `org.elasticsearch.xpack.security.operator.Constant`

Reply:
Thanks Jim! I've implemented IndicesRequest in the Request. I'd like some advice on the `IndicesOptions` - I guess we don't want any expansions, so I've set to `false` all options. Is there some better way to do this, or should we allow any of the `IndicesOptions`?

Reply:
I added a comment here: https://github.com/elastic/elasticsearch/pull/93975#discussion_r1115580241